sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -96,13 +95,15 @@ trait BaseLimitExec extends UnaryExecNode with CodegenSupport {
  */
 case class LocalLimitExec(limit: Int, child: SparkPlan) extends BaseLimitExec {
   override def outputOrdering: Seq[SortOrder] = child.outputOrdering
+  override def outputPartitioning: Partitioning = child.outputPartitioning
 }
 
 /**
  * Take the first `limit` elements of the child's single output partition.
  */
 case class GlobalLimitExec(limit: Int, child: SparkPlan) extends BaseLimitExec {
   override def requiredChildDistribution: List[Distribution] = AllTuples :: Nil
+  override def outputPartitioning: Partitioning = child.outputPartitioning
```

Comment:
This should be `SinglePartition` since `requiredChildDistribution` is `AllTuples`.